### ISSUE TITLE
Yolov8 won't detect yaml file

### ISSUE THREAD

**Issue by Harri1118** [ultralytics/ultralytics#2390]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I keep attempting to make my yolov8 script detect a yaml file, here is my code:

from ultralytics import YOLO
model = YOLO("yolov8n.yaml")

results = model.train(data="config.yaml", epochs=1)

Here is the error I receive:

Ultralytics YOLOv8.0.92  Python-3.10.11 torch-2.0.0+cpu CPU
yolo\engine\trainer: task=detect, mode=train, model=yolov8n.yaml, data=config.yaml, epochs=1, patience=50, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=False, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=0, resume=False, amp=True, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_thickness=3, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, v5loader=False, tracker=botsort.yaml, save_dir=runs\detect\train25
Traceback (most recent call last):
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\engine\trainer.py", line 122, in __init__
    self.data = check_det_dataset(self.args.data)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\data\utils.py", line 195, in check_det_dataset
    data = check_file(dataset)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\utils\checks.py", line 292, in check_file
    raise FileNotFoundError(f"'{file}' does not exist")
FileNotFoundError: 'config.yaml' does not exist

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Users\harri\Documents\pythonAlgorithms\yolov8_training\main.py", line 4, in <module>
    results = model.train(data="config.yaml", epochs=1)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\engine\model.py", line 365, in train
    self.trainer = TASK_MAP[self.task][1](overrides=overrides, _callbacks=self.callbacks)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\engine\trainer.py", line 126, in __init__
    raise RuntimeError(emojis(f"Dataset '{clean_url(self.args.data)}' error ❌ {e}")) from e
RuntimeError: Dataset 'config.yaml' error  'config.yaml' does not exist

Is there any way to fix this? I'm out of ideas at this point

**Comment by Laughing-q** [comment]
@Harri1118 looks like your `config.yaml` doesn't exist, please pass the valid path.

**Comment by Harri1118** [comment]
The issue is that my config.yaml file does exist and is in the same directory as the python script. I have attempted to put the full path in the line
results = model.train(data="config.yaml", epochs=1)
But still the script doesn't detect my file

**Comment by Laughing-q** [comment]
@Harri1118 try an absolute path.

**Comment by Harri1118** [comment]
This error is returned:

optimizer: SGD(lr=0.01) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias
train: Scanning C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\labels\train... 0 images, 15 backgrounds, 0 corrupt: 100%|██████████| 15/15 [00:00<00:0
train: WARNING  No labels found in C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\labels\train.cache. See https://docs.ultralytics.com/yolov5/tutorials/train_custom_data
train: New cache created: C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\labels\train.cache
Traceback (most recent call last):
  File "c:\Users\harri\Documents\pythonAlgorithms\yolov8_training\main.py", line 4, in <module>
    results = model.train(data=r"C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\config.yaml", epochs=1)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\engine\model.py", line 370, in train
    self.trainer.train()
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\engine\trainer.py", line 191, in train
    self._do_train(world_size)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\engine\trainer.py", line 268, in _do_train
    self._setup_train(world_size)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\engine\trainer.py", line 250, in _setup_train
    self.train_loader = self.get_dataloader(self.trainset, batch_size=batch_size, rank=RANK, mode='train')
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\v8\detect\train.py", line 59, in get_dataloader
    dataset = self.build_dataset(dataset_path, mode, batch_size)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\v8\detect\train.py", line 33, in build_dataset
    return build_yolo_dataset(self.args, img_path, batch, self.data, mode=mode, rect=mode == 'val', stride=gs)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\data\build.py", line 74, in build_yolo_dataset
    dataset = YOLODataset(
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\data\dataset.py", line 39, in __init__
    super().__init__(*args, **kwargs)
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\data\base.py", line 69, in __init__
    self.labels = self.get_labels()
  File "C:\Users\harri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\ultralytics\yolo\data\dataset.py", line 123, in get_labels
    raise FileNotFoundError(f'{self.prefix}No labels found in {cache_path}, can not start training. {HELP_URL}')
FileNotFoundError: train: No labels found in C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\labels\train.cache, can not start training. See https://docs.ultralytics.com/yolov5/tutorials/train_custom_data

**Comment by Laughing-q** [comment]
@Harri1118 it says you don't have labels, only images.

**Comment by Harri1118** [comment]
My labels directory is in C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\labels.

How can I make it so that the program detects the labels directory?

**Comment by Laughing-q** [comment]
@Harri1118 show me your config.yaml and path of training images and labels.

**Comment by Harri1118** [comment]
My config file:

path: C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data
train: images\train
val: images\train

names:
  0: dog

Images path: C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\images\train
labels path: C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\labels

**Comment by Laughing-q** [comment]
@Harri1118 you have to modify your labels path to `C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\labels\train` since your images path is `C:\Users\harri\Documents\pythonAlgorithms\yolov8_training\data\images\train`. You should keep them consistent.

**Comment by trnikon** [comment]
I stopped tuning early (ctrl+c) because the number of iterations I put was too high and it was taking too long and I noticed that tuning only saved the last weights, not the best weights. I searched manually for the best training session because I got no reference to it in the console output and `_tune/evolve.csv` had no reference to the trial number and used a different metric than `top1_acc` called `fitness_score` whose top value indicated a different trial. I tried loading the configuration of the best training session from the `args.yaml` file in order to retrain using them but I couldn't find documentation on it and was unsuccessful.
